Feature dev

DemevengDerrick opened the pull request:
Hi @truenomad, I added a function to get all data endpoints from a user profile. The return of this function is a table containing all form names, title s and ids. I just copied and twirked your initial function to produce this one. Can we abstract the **check_status_api** and **get_ona_page**? These functions will be regurlarly used for other ONA endpoints and i am not sure it is efficient to be copying it over and over.

DemevengDerrick committed:
Added handling POSIXct and POSIXT datetypes to prep_match_datatypes

DemevengDerrick committed:
Added a new function to get ONA data endpoints

truenomad commented:
Hi @DemevengDerrick 

Great addition! I like this additional function as we can use it each time before pulling the ONA data. I have simplified it more, so that the API status check is done inside this prep_ona_data_endpoints function. Now `prep_ona_data_endpoints` is used inside `get_ona_data` function before any data fetching happens, and the table is evaluated to see if it contains our form_id. If it doesn't it aborts!

truenomad committed:
Refine  and abstract`prep_ona_data_endpoints`

Co-Authored-By: Demeveng Derrick <46376095+DemevengDerrick@users.noreply.github.com>

truenomad committed:
Creates `get_multi_ona_data` (#10)

This solves #10 and produces `get_multi_ona_data` function for users to download multiple ONA data into a single dataframe.
